reporchestrator.py

```diff
@@ -1,65 +1,304 @@
-                                                   branch.commit),
-                          head=True, author=developer, committer=developer,
-                          author_date=date, commit_date=date)
-        master.commit = repo.head.commit
-        repo.head.reference = master
-        repo.delete_head(branch_ticket)
-        branch_commits = 0
-        branch_ticket = None
-    if branch_ticket is None:
-        branch_ticket = "ACME-" + str(randint(1, REPO_AGE_IN_DAYS))
-        branch_developer = (branch_developer + 1) % len(DEVELOPERS)
-        branch = repo.create_head(branch_ticket)
-        repo.head.reference = branch
-        repo.head.reset(index=True, working_tree=True)
-        branch_planned_commits = randint(1, MAX_COMMITS_PER_BRANCH)
\ No newline at end of file
+DEVELOPER_STRATEGIES = ("random-uniform", "round-robin")
+
+
+def pairs_to_actors(pairs):
+    """Transform a list of pairs to a list of actors."""
+    return [Actor(*pair) for pair in pairs]
+
+
+def activate_model(cfg):
+    """Activate the dynamic parts."""
+    cfg["fake"] = cfg["fake"]()
+    return cfg
+
+
+def seed_model(cfg):
+    """Seed the randomness."""
+    cfg["fake"].seed(cfg["random_seed"])
+    random.seed(cfg["random_seed"])
+    cfg["random_state"] = random.getstate()
+    return cfg
+
+
+def json_from_file(name):
+    """System call in one place ..."""
+    with open(name) as f_p:
+        return json.load(f_p)
+
+
+def mixin(argv=None):
+    """External configuration mixin load."""
+    mixin_cfg = {}
+    if not argv:
+        default_config_path = __file__.replace(".py", ".json")
+        if os.path.isfile(default_config_path):
+            mixin_cfg = json_from_file(default_config_path)
+    elif len(argv) == 1:
+        cfg_path_or_text = argv[0]
+        if os.path.isfile(cfg_path_or_text):
+            mixin_cfg = json_from_file(cfg_path_or_text)
+        elif cfg_path_or_text:
+            mixin_cfg = json.loads(cfg_path_or_text)
+
+    if mixin_cfg and mixin_cfg.get("developer_data"):
+        mixin_cfg["developers"] = pairs_to_actors(mixin_cfg["developer_data"])
+
+    return mixin_cfg
+
+
+def defaults(argv=None):
+    """Layered fill-in of default parameters - any errors not caught."""
+    default_cfg = {
+        "random_seed": 42,
+        "repo_age_in_days": 10,
+        "fake": Faker,
+        "team_size": 3,
+        "developer_strategy": "random-uniform",
+        "general_commit_words": ["Add", "an", "empty", "change"],
+        "merge_commit_words": ["Introduce", "the", "feature"],
+        "max_commits_per_branch": 10,
+        "repo_dir": "repository",
+        "datetime_format_template": r"%Y-%m-%dT%H:%M:%S",
+        "ticket_id_template": r"ACME-%d",
+        "message_template": r"%s %s",
+    }
+    mixin_cfg = mixin(argv)
+    cfg = {**default_cfg, **mixin_cfg}
+
+    cfg = activate_model(cfg)
+    cfg = seed_model(cfg)
+
+    cfg["developer_data"] = [
```

Comment:
We should not overwrite any mixin contribution to `developer_data` and `developers`, but maybe check for consistency only (length and `team_size`